core/templates/components/state-editor/state-responses-editor/state-responses.component.ts

```diff
@@ -151,16 +151,11 @@ export class StateResponsesComponent implements OnInit, OnDestroy {
   }
 
   isSelfLoopThatIsMarkedCorrect(outcome: Outcome): boolean {
-    if (!outcome ||
-        !this.stateEditorService.getCorrectnessFeedbackEnabled()) {
-      return false;
-    }
-
     let currentStateName = this.stateName;
 
     return (
-      (outcome.dest === currentStateName) &&
-      outcome.labelledAsCorrect);
+      (outcome?.dest === currentStateName) &&
```

Comment:
Why is the question mark needed? The outcome should be provided. If it is not then something is wrong and that should be investigated properly.

Reply:
@seanlip the original condition was ```if (!outcome || !this.stateEditorService.getCorrectnessFeedbackEnabled()) { return false; }``` and in testing, this if block was hit due to correctness feedback being true. If we add a separate check for null or undefined outcome, then we have to add a test with null and undefined outcome, which I believe is unnecessary just for the sake of test coverage. The current ?. (optional chaining) also gives the same output if the outcome is not provided, i.e., return false.

Reply:
Yup this seems fine -- but please add a test for null outcome if one is missing. This is important to ensure the code continues to work correctly in the future.

Reply:
Done.